common/build/eslint-plugin-fluid/src/rules/no-unchecked-record-access.js

```diff
@@ -39,18 +39,28 @@ module.exports = {
 
 		// Main function to run on every member access (e.g., obj.a or obj["a"])
 		function checkPropertyAccess(node) {
-			// Only check index signature types
+			const fullName = getFullName(node);
+
 			if (!isIndexSignatureType(parserServices, node)) {
 				return;
 			}
 
-			// Skip if the property has been checked (e.g., with optional chaining). Please see isDefined() for exhaustive list.
-			if (propertyHasBeenChecked(node)) {
-				return;
+			const parentNode = node.parent;
+			if (parentNode.type === "MemberExpression") {
```

Comment:
Can we document what this is doing? It isn't immediately clear to me, so I suspect it won't be for others.

Reply:
I was able to remove this since propertyHasBeenChecked should handle this recursively